### ISSUE TITLE
Zone Sensor Duplicates

### ISSUE THREAD

**Issue by garyak** [PeteRager/lennoxs30#10]
The screenshot shows duplicate instances of the North zones. I've captured a log file as requested. The file is not sanitized, so would prefer to send to you directly. 
02f6a2 is Master/North_Master. Inconsistent naming of the Thermostat and Zones itself may be the cause of your code generating two sensors. 
 
![lennoxdup](https://user-images.githubusercontent.com/47036329/120940453-9d59a480-c6da-11eb-8af1-8364437bad16.jpg)

**Comment by PeteRager** [comment]
thanks - send to here
pete dot sage at icloud dot com

**Comment by PeteRager** [comment]
i haven't forgotten about this, next on my list.

**Comment by garyak** [comment]
This might be the HomeKit Controller integration I've been using. I'm working to clear it from my instance now that your component is functional.

**Comment by PeteRager** [comment]
Those first two look like HomeKit.  I have that integration running also and it always reports "3" for supported features.  This integration has a hard-coded 27 and it's naming convention is <system_name>_<zone_name>

That said, in the log file I see two separate systems.

One system is called:
    North and has zones 
          Master
          Office
          Media Equipment
          Media Room
    South and has zones
           Zone 1
           Zone 2 - Zone 4 appear unconfigured

Is this correct? 

What is the physical layout?  E.g. one HVAC system with 5 zones?  Two S30 Panels?  3 remote zone sensors?

So we should have 5 climate sensors; but I suspect you are only seeing the first 4?

**Comment by garyak** [comment]
Yes. North has 4 zones, one s30 and 3 remote thermostats. South just one zone controlled by a single s30. Two split HVAC systems. Five climate sensors total.

**Comment by PeteRager** [comment]
Thanks.   I see the problem.  The initialization waits around until it finds a system with a zone; as it takes an indeterminate amount of time for the messages to arrive.  In your case it needs to wait for zones to appear on all system. 

 I'll let you know when I have an update.

**Comment by PeteRager** [comment]
I believe I've fixed this issue.  Grab a new copy of the branch

https://github.com/PeteRager/lennoxs30/tree/2021.06.11

If for some reasons it doesn't work, search around in the log for entries with this:
__init__:async_setup

and grab a bit around it.

I've also added support for heat and cool mode.  I can't test this (I have 1 zone AC), but I was able to see what the command is from the debug log you sent.  So it **should** work.
